### ISSUE TITLE
Warning Message: 1: In mean.default(cv.errs) : argument is not numeric or logical: returning NA

### ISSUE THREAD

**Issue by aennecken** [lutsik/MeDeCom#9]
Hello,
I ran Medecom on 4 samples of mixed lung tissue (containing tumour and healthy tissue). The data consists of EPIC arrays. I calculated b-Vals with minfi, selected the 25.000 most variant probes, and ran MeDeCom on a SGE system using the following command:
qsub qsub_MeDeCom.txt Medecom_mix3.txt

[qsub_MeDeCom.txt](https://github.com/lutsik/MeDeCom/files/1883279/qsub_MeDeCom.txt)
[Medecom_mix3.txt](https://github.com/lutsik/MeDeCom/files/1883285/Medecom_mix3.txt)

The script finishes and creates the desired Rdata output file. However I receive a warning as well. Here is a part of my log file:
[SGE jobs:] 11 remaining
[SGE jobs:] 4 remaining
[SGE jobs:] 4 remaining
[SGE jobs:] 4 remaining
[Main:] finished all jobs. Creating the object
There were 46 warnings (use warnings() to see them)
Warning messages:
1: In mean.default(cv.errs) : argument is not numeric or logical: returning NA
2: In mean.default(cv.errs) : argument is not numeric or logical: returning NA
3: In mean.default(cv.errs) : argument is not numeric or logical: returning NA
4: In mean.default(cv.errs) : argument is not numeric or logical: returning NA

When I tried to continue I received an error due to the NAs in the data set.
1: In max(vals[!is.na(vals)]) : no non-missing arguments, returning NA

Do you have an explanation for this warning? Is there a minimum number of samples that are required for a MeDeCom Analysis?
Thanks a lot for your reply and help!
Cheers,
Anne

**Comment by nikitaved** [comment]
Dear Anne,

Sorry for that super late reply. Looks like there might be a problem in the CPP solver.
I am not an expert in R, but is it possible for me to somehow reproduce your experiments?

Best,
Nik

**Comment by aennecken** [comment]
Dear Nik,
thanks a lot for your reply. In the meantime I also ran Medecom with the example data in the R package on our SGE and unfortunately received the same warning as stated above. As it ran fine locally on a subset I guess it must be a problem with the SGE. Could you suggest me how I could find out if the CPP solver might be the problem? 
Thanks for your reply! Anne

**Comment by nikitaved** [comment]
Ah, I see. So, no NAs while running locally, right? If this is the case, then, I guess, the solver should be fine, at least for this dataset =) (I am not sure that the solver is bug-free for some exotic border cases).
Otherwise, indeed, the problem is in the cluster-related implementation...

**Comment by aennecken** [comment]
Yes, the output was fine when I can it locally. I uploaded the scripts I used to qsub the jobs (see first post). If you have any troubleshooting ideas that would be great? I never had any problems with submitting Rscripts via qsub here.

**Comment by nikitaved** [comment]
Well, unfortunately, it is not my domain - I was responsible for the solver.
I guess, the answer to the question can only give @lutsik .
